File: 3241-divide-array-into-arrays-with-max-difference/3241-divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int n = nums.size(), parts = k/n;
        vector<vector<int>> ans;
        for(int i = 0;i<n;i+=3){
            if(nums[i+2]-nums[i] > k)return {};
            vector<int> tmp(nums.begin() + i, nums.begin() + i + 3);
            ans.push_back(tmp);
        }
        return ans;
    }
    };

File: 3241-divide-array-into-arrays-with-max-difference/3241-divide-array-into-arrays-with-max-difference.cs
public class Solution {
    public int[][] DivideArray(int[] nums, int k) {
        Array.Sort(nums);
        int n = nums.Length;
        List<int[]> ans = new List<int[]>();
        for(int i=0;i<n;i+=3){
            if(nums[i+2] - nums[i] > k)return new int[0][];
            ans.Add(new int[] {nums[i] , nums[i+1] , nums[i+2]});
        }
        return ans.ToArray();
    }
}